### PR TITLE
Adding representations, integration test for representations, and updated readme for auth types

### PR DESCRIPTION
**Pull request by carycheng77**
Let me know if there is anything I should change!

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @carycheng77 has signed the CLA. Thanks for the pull request!

File: Box.V2/Managers/BoxFilesManager.cs [hunk 1]
```diff
@@ -1113,6 +1113,36 @@ private int GetTimeDelay(HttpResponseHeaders headers)
...
+			BoxRequest request = new BoxRequest(_config.FilesEndpointUri, id)
+				.Method(RequestMethod.Get)
+				.Header(Constants.RequestParameters.XRepHints, representation)
+                .Header(Constants.RequestParameters.ContentDispositionType, contentDispositionType)
+                .Header(Constants.RequestParameters.ContentDispositionFilename, contentDispositionFilename)
+				.Param(ParamFields, representationsField);
+
+			IBoxResponse<BoxFile> response = await ToResponseAsync<BoxFile>(request).ConfigureAwait(false);
+
+			return response.ResponseObject.Representations;
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: Is there an use case that the representation is not ready immediately?

File: README.md [hunk 2]
```diff
@@ -29,6 +29,26 @@ PM> Install-Package Box.V2.Core
 
 If you haven't already created an app in Box go to https://developer.box.com/ and click 'Sign Up'
 
+### Deciding on Type of Authentication
+
+#### Build a Box Integration (Oauth 2.0)
```
Comment (coolboy) [inline README.md]: How does this relates to the rep API update?

File: Box.V2/Managers/BoxFilesManager.cs [hunk 3]
```diff
@@ -1113,6 +1113,36 @@ private int GetTimeDelay(HttpResponseHeaders headers)
...
+		/// for all other representation types.
+		/// </summary>
+		/// <param name="id">Id of the file (Required).</param>
+		/// <param name="RepresentationType">Enum value of representation requested or string of representation requested (Required).</param>
+		/// <param name="contentDispositionType"> Optional value set to "inline" or "attachment" 
+		/// <param name="contentDispositionFilename"> Optional value to define the downloaded representation's file name.</param>
+		/// <returns>A full file object containing the updated representations template_url and state is returned.</returns>
+		/// </summary>
+		public async Task<BoxRepresentationCollection<BoxRepresentation>> GetRepresentationsAsync(string id, string representation,
+            string contentDispositionType = null, string contentDispositionFilename = null)
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: Why not a complex request type like 
public async Task<BoxPreflightCheck> PreflightCheck(BoxPreflightCheckRequest preflightCheckRequest)

Would like to hear your reason here

File: Box.V2/Config/Constants.cs [hunk 4]
```diff
@@ -220,11 +220,58 @@ public static class RequestParameters
             public const string Pending = "pending";
 
             public const string IfMatch = "If-Match";
+
+            /*** Values for specifically representations endpint ***/
```
Comment (coolboy) [inline Box.V2/Config/Constants.cs]: endpoint

File: Box.V2/Managers/BoxFilesManager.cs [hunk 5]
```diff
@@ -1113,6 +1115,40 @@ private int GetTimeDelay(HttpResponseHeaders headers)
...
+        /// <returns>A full file object containing the updated representations template_url and state is returned.</returns>
+        /// </summary>
+        public async Task<BoxRepresentationCollection<BoxRepresentation>> GetRepresentationsAsync(BoxRepresentationRequest representationRequest)
+        {
+            representationRequest.ThrowIfNull("representationRequest")
+                .FileId.ThrowIfNullOrWhiteSpace("representationRequest.FileId");
+
+            BoxRequest request = new BoxRequest(_config.FilesEndpointUri, representationRequest.FileId)
+                .Method(RequestMethod.Get)
+                .Header(Constants.RequestParameters.XRepHints, representationRequest.XRepHints)
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: did you test they cases that some headers are missing?

File: Box.V2/Managers/BoxFilesManager.cs [hunk 6]
```diff
@@ -1113,6 +1115,40 @@ private int GetTimeDelay(HttpResponseHeaders headers)
...
+
+            IBoxResponse<BoxFile> response = await ToResponseAsync<BoxFile>(request).ConfigureAwait(false);
+
+            if (response.Status == ResponseStatus.Success)
+            {
+                return response.ResponseObject.Representations;
+            }
+            else
+            {
+                return null;
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: When are we going to have the null case? Is it 202 Accepted from the server?

File: Box.V2/Config/Constants.cs [hunk 7]
```diff
@@ -220,11 +220,59 @@ public static class RequestParameters
             public const string Pending = "pending";
 
             public const string IfMatch = "If-Match";
+
+            /*** Values for specifically representations endpoint ***/
+            public const string XRepHints = "x-rep-hints";
+            public const string SetContentDispositionType = "set_content_disposition_type";
+            public const string SetContentDispositionFilename = "set_content_disposition_filename";
+            public const string RepresentationField = "representations";
+            public const int RepresentationRequestRetryTime = 3000;
```
Comment (coolboy) [inline Box.V2/Config/Constants.cs]: make it private inside the FilesManager, since this is only related to the retry but not the request

**Pull request merged**